core/src/main/scala/kanaloa/reactive/dispatcher/Regulator.scala

```diff
@@ -118,14 +118,20 @@ object Regulator {
     weightOfLatestMetric:   Double         = 0.5
   )
 
+  private val MaximumEstimatedDelayRatio = 100000
+
   private def estimateDelay(
-    queueLength: QueueLength,
-    speed:       Speed
-  ): Option[FiniteDuration] =
-    if (speed.value == 0) None else Some((queueLength.value.toDouble / speed.value).milliseconds)
+    queueLength:    QueueLength,
+    speed:          Speed,
+    referenceDelay: FiniteDuration
+  ): Option[FiniteDuration] = {
+    val totalMillis = queueLength.value.toDouble / speed.value
+    //avoid creating too long an estimated delay, which is meaningless anyway
+    if (speed.value == 0 || totalMillis > (referenceDelay.toMillis * MaximumEstimatedDelayRatio)) None else Some(totalMillis.milliseconds)
```

Comment:
the `None` value is indicating that delay is too long to estimate (note it's None when the speed is zero) . I thought about caping it at the max, but that might give some false impression on grafana chart. 
I am adding a test.